packages/db/src/lib/file.ts

```diff
@@ -49,5 +69,30 @@ export const getFilesAndGranuleInfoQuery = ({
   if (limit) {
     query.limit(limit);
   }
+  if (collectionIds.length > 0) {
+    query.innerJoin(
+      collectionsTable,
+      `${granulesTable}.collection_cumulus_id`,
+      `${collectionsTable}.cumulus_id`
+    );
+    const collectionIdConcatField = `(${collectionsTable}.name || '${collectionIdSeparator}' || ${collectionsTable}.version)`;
```

Comment:
This was my suggestion to Matt, but perhaps it wasn't a good one. I think my initial concern was whether PG would interpret the query correctly. The goal here is to filter files by collections through granules. 

Doing a `(collections.name___collections.version) IN ('a___1', 'b___2')` will definitely restrict the results to only the desired collections.

If you use an OR where, I'm wondering if it would be the same, because I think the query will look like

```
OR (name = 'a' AND version = '1') OR (name = 'b' and version '2')
```

Perhaps the effect is the same. Easy enough to test. If the effect is the same then I agree that @Jkovarik's method is much better

Reply:
This is good information.  number of collections are selected by operator, I would hope fewer than 10 or 20 at most.  But it's available on the api so a developer could in theory do more.

This does a combo setup because I originally was doing this in two steps where I found the collection_cumulus_ids first and then used `query.whereIn('collection_cumulus_id', collectionCumulusIds)` 

I think Jonathan's suggestion will give the same result.  I'll check.

Reply:
Tidying up a bit, I *think* what we want is something like: 

```
    const nameVersionPairs = collectionIds.map((id) => deconstructCollectionId(id));
    // eslint-disable-next-line func-names
    query.andWhere(function () {
      this.where('collections', nameVersionPairs.pop());
      nameVersionPairs.forEach((pair) => this.orWhere('collection', pair));
    });
 ```

Reply:
Paired on this, and this should work: 


```

    const nameVersionPairs = collectionIds.map((id) => deconstructCollectionId(id));
    // eslint-disable-next-line func-names
    query.andWhere(function () {
      const innerQuery = this;
      const firstPair = nameVersionPairs.pop() as { name: string, version: string };
      innerQuery.where(firstPair);
      nameVersionPairs.forEach((pair) => innerQuery.orWhere(pair));
    });

```

Reply:
8276eae959

Reply:
well, that and move the first pair up out of the local function scope (found after pairing again)